fedimint-server/src/config/mod.rs

```diff
@@ -449,28 +452,20 @@ impl ServerConfig {
         // of each module that was compiled in. This is how things were
         // initially, where we consider "module as a code" as "module as an instance at
         // runtime"
-        for (module_instance_id, (_kind, gen)) in
-            module_config_gens.legacy_init_order_iter().enumerate()
-        {
+        for (module_instance_id, (_kind, gen)) in registry.legacy_init_order_iter().enumerate() {
             let module_instance_id = u16::try_from(module_instance_id)
                 .expect("64k module instances should be enough for everyone");
             module_cfgs.insert(
                 module_instance_id,
-                if let Ok(cfgs) = gen
-                    .distributed_gen(
-                        connections,
-                        our_id,
-                        module_instance_id,
-                        peers,
-                        &params.modules,
-                        task_group,
-                    )
-                    .await?
-                {
-                    cfgs
-                } else {
-                    return Ok(Err(Cancelled));
```

Comment:
We used to this custom handling, so a canceled operation is not considered an error and user pressing ctrl+c does not lead to panics and/or screen full of errors.

Reply:
We assume a failure in the middle of a DKG run is a cancellation?  Doesn't this swallow the underlying error if there is one?

Reply:
Can we make a determination whether or not to display an error based on the `TaskGroup`?

Reply:
`Cancellable<Result<X>>` makes a distinction between operation being cancelled via ctrl+c : `Err(Cancelled)` or hitting an error `OK(Err(...))`. If it was cancelled we are not supposed to print errors, etc. If it failed, we handle like normally we treat errors.

It's a bit inconvenient, because we have to "unpeel" the inner error in calling places, but it forces handling it correctly, and there isn't many places we need to take care of it.

Reply:
But as far as I can tell we just return `Cancelled` if `run_g1` ever returns an error, regardless of whether this was a cancellation or an actual error with DKG.

Reply:
Perhaps we should spawn every instance of DKG as a separate subtask and swallow the errors if the task is cancelled?

Reply:
> But as far as I can tell we just return Cancelled if run_g1 ever returns an error, regardless of whether this was a cancellation or an actual error with DKG.

Possibly it got broken during recent refactoring, or I didn't do it correctly there when I first introduced it.

> Perhaps we should spawn every instance of DKG as a separate subtask and swallow the errors if the task is cancelled?

Swallowing errors leads to hard to debug issues when actual errors actually happen. Treating cancellation as an error leads to bad UX.

The only way to handle it robustly is to make a distinction between cancellation and an error (which is what `Cancellable` is about) and handle it properly everywhere.

Reply:
I see...I think it's not working properly now, let me try to fix.